nvi-commons/src/main/java/no/sikt/nva/nvi/common/service/NviService.java

```diff
@@ -45,134 +35,122 @@ public static NviService defaultNviService() {
         return new NviService(defaultDynamoClient());
     }
 
-    //TODO: Remove JacocoGenerated when other if/else cases are implemented
-    @JacocoGenerated
-    public Optional<CandidateWithIdentifier> upsertCandidate(CandidateEvaluatedMessage evaluatedCandidate) {
-        if (!isExistingCandidate(evaluatedCandidate) && isNviCandidate(evaluatedCandidate)) {
-            return Optional.of(createCandidate(evaluatedCandidate));
-        } else if (isExistingCandidate(evaluatedCandidate) && isNviCandidate(evaluatedCandidate)) {
-            var existing = findByPublicationId(evaluatedCandidate.candidateDetails().publicationId()).orElseThrow();
+    @JacocoGenerated //TODO Temporary for coverage
+    public Optional<Candidate> upsertCandidate(DbCandidate candidate) {
+        if (!isExistingAndApplicableCandidate(candidate.publicationId()) && candidate.applicable()) {
+            return Optional.of(
+                createCandidate(candidate,
+                                generatePendingApprovalStatuses(candidate.points())
+                ));
+        } else if (isExistingAndApplicableCandidate(candidate.publicationId()) && candidate.applicable()) {
```

Comment:
You renamed the method to if `isExistingAndApplicableCandidate()` but are did not move the `candidate.applicable()` -check?

Reply:
Ops

Reply:
Btw can't move applicable inside without breaking the logic

Reply:
See this commen: https://github.com/BIBSYSDEV/nva-nvi/pull/66#discussion_r1316950412